src/api/characters.js

```diff
@@ -0,0 +1,18 @@
+import { prop } from 'ramda';
+import { client } from './tools';
+
+export const createCharacter = async character => {
+  const response = await client.post('/characters', { ...character })
+    .then(prop('data'))
```

Comment:
Is there a reason to mix the Promise-y `then` and `catch` in here when the function is already `async`?

Reply:
Hm probably not if it jumps out as weird. I confess that I'm not an expert on the best way to handle these API calls. How would you suggest we format these functions? Would we just wrap the call in a `try {} catch {}` instead?

```js
export const createCharacter = async character => {
  try {
    const response = await client.post('/charactersd', { ...character });

    return prop('data', response);
  } catch (err) {
    console.error(err);
  }
};
```

Reply:
Although, what we really want to do is catch the error at the action layer, so we can call the `characterCreationFailed` action instead of `characterCreated`.

Reply:
If this was Go, Rust, or TypeScript, then I would say a good pattern could be something like
```javascript
export const createCharacter = async character => {
  try {
    const response = await client.post('/charactersd', { ...character });

    return [prop('data', response), undefined];
  } catch (err) {
    console.error(err);
    return [undefined, err];
  }
};
```

What makes me hesitant is that JavaScript does not have tuples, and the "returns a value or an error" pattern is not something I have seen in plain JavaScript. That said, if we want to start proper error handling now, then we could set that pattern or something like it.

Reply:
That syntax should work fine in JS. That's how the `useState` hook works. Looks like this:

`[stateValue, valueSetterFunction] = useState(initialValue);`

However, I have a slightly different potential solution, based on my comment about handling errors in actions instead of at the API level. We could keep the API layer super simple:

```js
export const createCharacter = async character => {
  const response = await client.post('/characters', { ...character });

  return prop('data', response);
};
```

Then add error handling at the action layer:

```js
export const createCharacter = character => async (dispatch, getState) => {
  const playerId = getCurrentPlayerId(getState());

  const newCharacter = { ...character, playerId };

  dispatch(creatingCharacter);
  charactersApi.createCharacter(newCharacter)
    .then(newCharacter => {
      console.log(newCharacter);
      dispatch(characterLoaded(newCharacter));
      dispatch(navTo(GAME_ROUTE.replace(':gameId', character.gameId)));
    })
    .catch(err => dispatch(creatingCharacterFailed(err)));
};
```

Reply:
I like it!

Reply:
I'll still do the `try {} catch {}` syntax though.

```js
export const createCharacter = character => async (dispatch, getState) => {
  const playerId = getCurrentPlayerId(getState());

  const newCharacter = { ...character, playerId };

  dispatch(creatingCharacter);
  try {
    const createdCharacter = await charactersApi.createCharacter(newCharacter);

    dispatch(characterLoaded(createdCharacter));
    dispatch(navTo(GAME_ROUTE.replace(':gameId', character.gameId)));
  } catch (err) {
    dispatch(creatingCharacterFailed(err));
  }
};
```